ra/ra.go

```diff
@@ -172,37 +176,65 @@ func validateEmail(ctx context.Context, address string, resolver bdns.DNSResolve
 	}
 	splitEmail := strings.SplitN(emails[0].Address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
-	var resultMX []string
-	var resultA []net.IP
-	var errMX, errA error
-	var wg sync.WaitGroup
-	wg.Add(2)
+	probChan := make(chan *probs.ProblemDetails, 2)
+	timeoutCtx, _ := context.WithTimeout(ctx, ra.emailValidationTimeout)
+
+	/*
+	 * Note: we use `problemUnlessTimeout` here because we are OK with treating
+	 * a timeout during email validation as a non-error. We perform the email
+	 * address validation as a best-effort check for typos and not as a strong
+	 * control. This helps avoid spurious RPC timeouts in the WFE from slow DNS
+	 * lookups.
+	 * See https://github.com/letsencrypt/boulder/issues/2260 for more.
+	 */
 	go func() {
-		resultMX, errMX = resolver.LookupMX(ctx, domain)
-		wg.Done()
+		resultMX, errMX := resolver.LookupMX(timeoutCtx, domain)
+		// No error but an empty DNS response is a problem for MX lookups
+		if errMX == nil && len(resultMX) == 0 {
+			probChan <- probs.InvalidEmail(emptyDNSResponseDetail)
+		} else {
+			probChan <- problemUnlessTimeout(errMX)
+		}
 	}()
 	go func() {
-		resultA, errA = resolver.LookupHost(ctx, domain)
-		wg.Done()
+		_, errA := resolver.LookupHost(timeoutCtx, domain)
+		// Unlike the MX lookup we are OK with an empty A/AAAA response. See
+		// `email.only` mock in `bdns/mocks.go`
+		probChan <- problemUnlessTimeout(errA)
 	}()
-	wg.Wait()
 
-	if errMX != nil {
-		prob := bdns.ProblemDetailsFromDNSError(errMX)
-		prob.Type = probs.InvalidEmailProblem
-		return prob
-	} else if len(resultMX) > 0 {
+	select {
```

Comment:
I think this select becomes unnecessary if we are relying on the DNSResolver to timeout.